### PR TITLE
[PE-1346] Additional intent order testing

### PR DESCRIPTION
**Pull request by prateekdefi**
Added intent order tests for the following scenarios:
- Open/close two intents in the same version such that there's a non-zero notional leftover (maker profit)
- Create a maker order and intent order to check fee exclusion works properly

### CODE REVIEW

File: packages/perennial/test/integration/core/fees.test.ts [hunk 1]
```diff
@@ -2385,4 +2389,200 @@ describe('Fees', () => {
...
+        ...DEFAULT_GUARANTEE,
+        orders: 1,
+        takerPos: POSITION.div(2),
+        takerNeg: POSITION.div(2),
+        takerFee: POSITION.div(2),
+      })
+      expectGuaranteeEq(await market.guarantees(user.address, (await market.locals(user.address)).currentId), {
+        ...DEFAULT_GUARANTEE,
+        orders: 1,
+        notional: BigNumber.from(569414885),
```
Comment (EdNoepel) [inline packages/perennial/test/integration/core/fees.test.ts:2469]: Two concerns:
- Number should be expressed as `parse6decimal('569.414885')` for readability.
- Comments should explain where the value came from and why it is correct.

Same with lines 2524, 2546, 2547, and 2569.

File: packages/perennial/test/integration/core/fees.test.ts [hunk 2]
```diff
@@ -2385,4 +2389,200 @@ describe('Fees', () => {
...
+      })
+
+      await chainlink.next()
+
+      await market.settle(user.address)
+      await market.settle(userB.address)
+      await market.settle(userC.address)
+      await market.settle(userD.address)
+
+      expectIntentEq(intent, user.address, userC.address, market)
```
Comment (EdNoepel) [inline packages/perennial/test/integration/core/fees.test.ts:2497]: Probably need a comment here to explain what we're trying to prove.  We have a market with 4 settled positions.  How do lines 2498-2587 show that the intent order did not influence fees?

File: packages/perennial/test/unit/market/Market.test.ts [hunk 3]
```diff
@@ -22064,6 +22070,477 @@ describe('Market', () => {
               liquidationFee: { _value: -riskParameter.liquidationFee.mul(SETTLEMENT_FEE).div(1e6) },
             })
           })
+
+          it('fills two positions and settles later with fee (above / long)', async () => {
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:22074]: What does `above / long` mean here and in the name of the next test?

File: packages/perennial/test/unit/market/Market.test.ts [hunk 4]
```diff
@@ -23648,6 +24131,203 @@ describe('Market', () => {
               liquidationFee: { _value: -riskParameter.liquidationFee.mul(SETTLEMENT_FEE).div(1e6) },
             })
           })
+
+          it('fills two positions and settles later with fee (taker / short)', async () => {
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:24135]: What does `taker / short` mean here?

File: packages/perennial/test/unit/market/Market.test.ts [hunk 5]
```diff
@@ -24287,3 +24967,30 @@ describe('Market', () => {
     })
   })
 })
+
+export async function expectIntentEq(intent: Intent, taker: string, maker: string, market: Market) {
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:24971]: Even though they're really strings, recommend using `Address` type for readability.

File: packages/perennial/test/unit/market/Market.test.ts [hunk 6]
```diff
@@ -24287,3 +24967,30 @@ describe('Market', () => {
...
+  const makerLong = makerPosition.long
+  const makerShort = makerPosition.short
+
+  const takerOrder = await market.pendingOrders(taker, (await market.locals(taker)).latestId)
+  const makerOrder = await market.pendingOrders(maker, (await market.locals(maker)).latestId)
+
+  expect(takerOrder.longPos).to.equal(takerLong, 'Order:LongPos')
+  expect(takerOrder.shortPos).to.equal(takerShort, 'Order:ShortPos')
+
+  expect(makerOrder.longPos).to.equal(makerLong, 'Order:LongPos')
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:24986]: The string hints for maker and taker orders are the same.  Recommend naming them `MakerOrder:ShortPos` and such for uniqueness.  Same with `Intent:Amount` on lines below.

File: packages/perennial/test/unit/market/Market.test.ts (hunk 5 above)
Comment (prateekdefi) [reply packages/perennial/test/unit/market/Market.test.ts]: This function has been removed.

File: packages/perennial/test/integration/core/fees.test.ts (hunk 1 above)
Comment (prateekdefi) [reply packages/perennial/test/integration/core/fees.test.ts]: done

File: packages/perennial/test/unit/market/Market.test.ts (hunk 6 above)
Comment (prateekdefi) [reply packages/perennial/test/unit/market/Market.test.ts]: This function has been removed.

File: packages/perennial/test/unit/market/Market.test.ts (hunk 3 above)
Comment (prateekdefi) [reply packages/perennial/test/unit/market/Market.test.ts]: for the test naming strings I have followed the same pattern as used everywhere else in this file. 
`above / long` means we are making long position and the price of intent is above the oracle price.

File: packages/perennial/test/unit/market/Market.test.ts (hunk 4 above)
Comment (prateekdefi) [reply packages/perennial/test/unit/market/Market.test.ts:24135]: `taker / short` here means that intent account already has short position and there is no position for intent filler.
`taker / long` means that the intent account already has long position and there is no position for intent filler.
`maker / short` means that the intent account has no position and there is already a short position for intent filler.
`maker / long` means that the intent account has no position and there is already a long position for intent filler.

File: packages/perennial/test/integration/core/fees.test.ts (hunk 2 above)
Comment (prateekdefi) [reply packages/perennial/test/integration/core/fees.test.ts]: [4ccb4ef](https://github.com/equilibria-xyz/perennial-v2/pull/440/commits/4ccb4ef11735696b0e716b3caefdfbef73102f9d)

File: packages/perennial/test/unit/market/Market.test.ts [hunk 7]
```diff
@@ -22334,12 +22330,23 @@ describe('Market', () => {
...
+              },
+            })
+
             const EXPECTED_PNL = parse6decimal('10') // position * (125-123)
             const TAKER_FEE = parse6decimal('6.15') // position * (0.01) * price
             const SETTLEMENT_FEE = parse6decimal('0.50')
 
             const intent = {
-              amount: -POSITION.div(2),
+              amount: POSITION.div(2),
```
Comment (EdNoepel) [inline packages/perennial/test/unit/market/Market.test.ts:22349]: Why was this made positive?  Original test context was `closes position`, but line 22317 was removed.

Comment (EdNoepel) [reply packages/perennial/test/unit/market/Market.test.ts:22349]: nvm; saw this `closes position` test moved down to line 22810

**Review by EdNoepel — APPROVED** [review]
Would benefit from a description sumarrizing the newly added tests.

**Review by kbrizzle — APPROVED** [review]

**Pull request merged**